exercises/error-handling/example.py

```diff
@@ -17,9 +17,8 @@ def handle_error_by_returning_tuple(input_data):
 
 
 def filelike_objects_are_closed_on_exception(filelike_object):
-    with filelike_object:
+    with filelike_object as fobj:
         try:
-            filelike_object.do_something()
-        except Exception:
-            filelike_object.close()
-            raise
+            fobj.do_something()
+        finally:
+            fobj.close()
```

Comment:
As I stated previously, you don't need both a context manager and a `try` block for this - they are two different methods of error handling. Either you use a context manager, which will call the `__enter__` and `__exit__` methods that will open and close the object, OR you handle opening and closing explicitly, using a `try ... finally` block to ensure that the object is closed once the `try` block has finished.

This function should either be:
```python
def filelike_objects_are_closed_on_exception(filelike_object):
    filelike_object.open()
    try:
        filelike_object.do_something()
    finally:
        filelike_object.close()
```

Or:
```python
def filelike_objects_are_closed_on_exception(filelike_object):
    with filelike_object as fobj:
        fobj.do_something()
```

Or:
```python
def filelike_objects_are_closed_on_exception(filelike_object):
    with filelike_object:
        filelike_object.do_something()
```

Reply:
Apologies, I had somehow managed to convince myself that when an exception is raised from within the context manager object itself, `__exit__()` would not necessarily be called correctly, thus the need for `try... finally` inside `with`. Fix forthcoming...